oprf/client.go

```diff
@@ -0,0 +1,156 @@
+package oprf
+
+import (
+	"crypto/rand"
+	"errors"
+)
+
+// Client is a representation of a Client during protocol execution.
+type Client struct{ suite }
+
+// ClientRequest is a structure to encapsulate the output of a Request call.
+type ClientRequest struct {
+	inputs          [][]byte
+	blinds          []Blind
+	BlindedElements []Blinded
+}
+
+// NewClient creates a new instantiation of a Client.
+func NewClient(id SuiteID, m Mode) (*Client, error) {
+	suite, err := suiteFromID(id, m)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{*suite}, nil
+}
+
+// Request generates a request for server passing an array of inputs to be
+// evaluated by server.
+func (c *Client) Request(inputs [][]byte) (*ClientRequest, error) {
+	if len(inputs) == 0 {
+		return nil, errors.New("few inputs")
+	}
+
+	blinds := make([]Blind, len(inputs))
+	for i := range inputs {
+		blinds[i] = c.suite.Group.RandomScalar(rand.Reader)
+	}
+
+	return c.blind(inputs, blinds)
+}
+
+func (c *Client) blind(inputs [][]byte, blinds []Blind) (*ClientRequest, error) {
+	var err error
+	blindedElements := make([]Blinded, len(inputs))
+	for i := range inputs {
+		p := c.suite.Group.HashToElement(inputs[i], c.suite.getDST(hashToGroupDST))
+		blindedElements[i], err = c.scalarMult(p, blinds[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &ClientRequest{inputs, blinds, blindedElements}, nil
+}
+
+// Finalize computes the signed token from the server Evaluation and returns
+// the output of the OPRF protocol. The function uses server's public key
+// to verify the proof in verifiable mode.
+func (c *Client) Finalize(r *ClientRequest, e *Evaluation, info []byte, pkS *PublicKey) ([][]byte, error) {
```

Comment:
So, in non-verifiable mode, `pkS` is ignored? Could we store it in `Client` instead?

Reply:
done